### PR TITLE
[1.1.19] mv hdfs file

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

EngineConn-Core defines the the abstractions and interfaces of the EngineConn core functions.
The Engine Service in Linkis 0.x is refactored, EngineConn will handle the engine connection 
and session management.

### Related issues/PRs

Related issues: #590
Related pr:#591

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Review by casionone — APPROVED** [review]

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 1]
```diff
@@ -1158,4 +1159,70 @@ private static void deleteAllFiles(FileSystem fileSystem, FsPath fsPath) throws
...
+        required = false,
+        dataType = "String",
+        value = "filePermission"),
+  })
+  @RequestMapping(path = "/chmod", method = RequestMethod.GET)
+  public Message chmod(
+      HttpServletRequest req,
+      @RequestParam(value = "filepath", required = false) String filePath,
+      @RequestParam(value = "isRecursion", required = false) Boolean isRecursion,
+      @RequestParam(value = "filePermission", required = false) String filePermission)
```
[the remark this answers is not in the document]
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java]: done

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 2]
```diff
@@ -1158,4 +1159,70 @@ private static void deleteAllFiles(FileSystem fileSystem, FsPath fsPath) throws
...
+      throws WorkSpaceException, IOException {
+    String userName = ModuleUserUtils.getOperationUser(req, "chmod " + filePath);
+    if (StringUtils.isEmpty(filePath)) {
+      return Message.error( MessageFormat.format(PARAMETER_NOT_BLANK, filePath));
+    }
+    if (StringUtils.isEmpty(filePermission)) {
+      return Message.error( MessageFormat.format(PARAMETER_NOT_BLANK, filePermission));
+    }
+    if (null == isRecursion) {
+      isRecursion = true;
```
[the remark this answers is not in the document]
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java:1199]: done

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java [hunk 3]
```diff
@@ -1158,4 +1161,72 @@ private static void deleteAllFiles(FileSystem fileSystem, FsPath fsPath) throws
...
+      return Message.error(MessageFormat.format(PARAMETER_NOT_BLANK, filePath));
+    }
+    if (StringUtils.isEmpty(filePermission)) {
+      return Message.error(MessageFormat.format(PARAMETER_NOT_BLANK, filePermission));
+    } else {
+      Matcher matcher = Pattern.compile("[0-7]{3}").matcher(filePermission);
+      if (matcher.matches()) {
+        return Message.error(MessageFormat.format(PERMISSION_FORMAT_ERROR, filePermission));
+      }
+    }
```
Comment (aiceflower) [inline linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java]: There's a slight problem with the logic. Please fix it.

**Review by aiceflower — COMMENTED** [review]
code review

File: linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java (hunk 3 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-script-dev/linkis-storage-script-dev-server/src/main/java/org/apache/linkis/filesystem/restful/api/FsRestfulApi.java]: done

**Review by casionone — APPROVED** [review]

**Pull request merged**